lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -223,6 +223,81 @@ class RecomposeSplitTensorGetItemOp
     return success();
   }
 };
+
+class RecomposeChunkListUnpack : public OpRewritePattern<PrimListUnpackOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(PrimListUnpackOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenChunkOp + PrimListUnpackOp to AtenSliceTensorOps
+    auto chunk = dyn_cast<AtenChunkOp>(op.getOperand().getDefiningOp());
+    if (!chunk)
+      return failure();
+    if (isListPotentiallyMutated(chunk.getResult()))
+      return failure();
+    Value dim = chunk.getDim();
+    Value input = chunk.getSelf();
+    Value chunks = chunk.getChunks();
+    Location loc = chunk.getLoc();
+    Value totalSize = rewriter.create<Torch::AtenSizeIntOp>(loc, input, dim);
+
+    // below code is equivalent to:
+    // if floordiv(totalSize, chunks) * chunks == totalSize:
+    //    chunkSize = floordiv(totalSize, chunks)
```

Comment:
I think it's no need to insert `PrimIfOp`, `chunkSize = floordiv(totalSize + chunks - 1, chunks)` is more simple.